### PR TITLE
Make HasCapability private

### PR DESCRIPTION
**Pull request by expertdicer**
## Description

closes: #3289 

### Commit Message / Changelog Entry

```bash
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/27-interchain-accounts/controller/keeper/export_test.go [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// SetParams sets the total set of params.
+func (k Keeper) HasCapability(ctx sdk.Context, portID string) bool {
```
Comment (crodriguezvega) [inline modules/apps/27-interchain-accounts/controller/keeper/export_test.go:8]: @expertdicer Is this public function really needed?

**Review by crodriguezvega — COMMENTED** [review]
Thank you, @expertdicer!

File: modules/apps/27-interchain-accounts/controller/keeper/export_test.go (hunk 1 above)
Comment (expertdicer) [reply modules/apps/27-interchain-accounts/controller/keeper/export_test.go:8]: Yes. It's for separating testing logic and core private logic. So in the future, if you want to modify in the testing layer, it won't affect the core private logic. I think so.

**Comment by expertdicer** [comment]
> Thank you, @expertdicer!

You're welcome!

File: modules/apps/27-interchain-accounts/controller/keeper/export_test.go (hunk 1 above)
Comment (DimitrisJim) [reply modules/apps/27-interchain-accounts/controller/keeper/export_test.go:8]: Seems like the documentation for this refers to something else?

Comment (expertdicer) [reply modules/apps/27-interchain-accounts/controller/keeper/export_test.go:8]: > Seems like the documentation for this refers to something else?

Sorry. I wasn't checking the comment carefully.

Comment (colin-axner) [reply modules/apps/27-interchain-accounts/controller/keeper/export_test.go:8]: This is a really cool construction. Never thought about wrapping private functions with a public testing only API. I do think I'd prefer if we removed these functions if they are unused. We can add them back if we find usage for them, but sometimes it's good to write tests in the same way external users will need to access information

Comment (crodriguezvega) [reply modules/apps/27-interchain-accounts/controller/keeper/export_test.go:8]: I agree with @colin-axner that if the function is not used at the moment, then better to just remove it for now.

**Review by crodriguezvega — CHANGES_REQUESTED** [review]
Thank you, @expertdicer. I think if we remove those testing functions and rebase with main, the PR should be ready to be merged. :)

**Review by crodriguezvega — CHANGES_REQUESTED** [review]
Thank you, @expertdicer. I think if we remove those testing functions and rebase with main, the PR should be ready to be merged. :)

File: modules/apps/27-interchain-accounts/controller/keeper/genesis.go [hunk 2]
```diff
@@ -12,9 +12,9 @@ import (
 // InitGenesis initializes the interchain accounts controller application state from a provided genesis state
 func InitGenesis(ctx sdk.Context, keeper Keeper, state genesistypes.ControllerGenesisState) {
 	for _, portID := range state.Ports {
-		if !keeper.HasCapability(ctx, portID) {
-			capability := keeper.BindPort(ctx, portID)
-			if err := keeper.ClaimCapability(ctx, capability, host.PortPath(portID)); err != nil {
+		if !keeper.hasCapability(ctx, portID) {
+			cap := keeper.BindPort(ctx, portID)
```
Comment (crodriguezvega) [inline modules/apps/27-interchain-accounts/controller/keeper/genesis.go:16]: Could you please rename this to `capability` to make the linter happy? :)

File: modules/apps/27-interchain-accounts/host/keeper/genesis.go [hunk 3]
```diff
@@ -12,9 +12,9 @@ import (
 
 // InitGenesis initializes the interchain accounts host application state from a provided genesis state
 func InitGenesis(ctx sdk.Context, keeper Keeper, state genesistypes.HostGenesisState) {
-	if !keeper.HasCapability(ctx, state.Port) {
-		capability := keeper.BindPort(ctx, state.Port)
-		if err := keeper.ClaimCapability(ctx, capability, host.PortPath(state.Port)); err != nil {
+	if !keeper.hasCapability(ctx, state.Port) {
+		cap := keeper.BindPort(ctx, state.Port)
```
Comment (crodriguezvega) [inline modules/apps/27-interchain-accounts/host/keeper/genesis.go:16]: Ditto about renaming to `capability`.

**Review by crodriguezvega — COMMENTED** [review]
We're almost there, @expertdicer! Just need to make the linter happy now. :)

File: modules/apps/27-interchain-accounts/controller/keeper/genesis.go (hunk 2 above)
Comment (expertdicer) [reply modules/apps/27-interchain-accounts/controller/keeper/genesis.go:16]: Ty

**Review by damiannolan — APPROVED** [review]
Thanks @expertdicer ❤️

File: modules/apps/27-interchain-accounts/host/keeper/export_test.go
[the remark this answers is not in the document]
Comment (expertdicer) [reply modules/apps/27-interchain-accounts/host/keeper/export_test.go:1]: i removed it with the HasCapability test. I don't know if there is a way to test this

**Pull request merged**